fapi-client/src/main/scala/com/gu/facia/api/http/HttpResponse.scala

```diff
@@ -21,11 +22,10 @@ object HttpResponse {
     else Response.Left(HttpError(s"Request failed with status code ${response.statusCode}"))
   }
 
-  def jsonResponse(request: Request, client: dispatch.Http)(implicit executionContext: ExecutionContext): Response[JValue] = {
+  def jsonResponse(request: Request, client: dispatch.Http)(implicit executionContext: ExecutionContext): Response[JsValue] = {
     for {
       rawResponse <- Right(client(request, HttpResponse.dispatchHandler))
       okResponse <- HttpResponse.okToRight(rawResponse)
-      json <- Json.toJson(okResponse.body)
-    } yield json
+    } yield Json.toJson(okResponse.body)
```

Comment:
Does this work the way you expect in the context of `Response`? The other one was throwing exceptions that were being handled by us.